### PR TITLE
char/healing: fix/rework healing rates to be more logical

### LINKED ISSUE #61202: healing_rate_medicine multiplies healing for every bandaged limb
### Describe the bug

I'm just going to include a pretty picture and a save for easy testing.
Untitled5

### Steps to reproduce

Load save, bandage and disinfect desired number of limbs, advance 6 hours and compare results to image.

### Expected behavior

My head wounds should not heal faster just because I have bandaged the rest of my broken body. If anything, they should heal *slower* because the rest of my body is hurt.

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10
- Game Version: 0.F-11529-gb93112f290-dirty [64-bit] (reporter note: compiled with extra debug messages ONLY)
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

### Additional context

[Powers.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/9626827/Powers.zip)

### PR DESCRIPTION
**Pull request by andrei8l**
#### Summary
None

#### Purpose of change
* The documentation, code, contributors, and mutation descriptions (and wiki!) all have different ideas of what `healing_awake` and `healing_resting` represent
* Sleeping characters get unjustified double healing
* The code for healing rate is unnecessarily hard to parse
* Healing effects on some limbs affect healing rate on other limbs as well. 
* Fixes: #61202

#### Describe the solution
Standardize the meaning of `healing_awake/resting`, namely:
- Rename `healing_resting`  to `healing_multiplier` and make it multiply the base healing rate
- Refluff/repurpose `healing_awake` to be a percentage of healing rate applied while awake
- Rewrite `Character::healing_rate()` to remove superfluous branching
- Modify healing mutations to maintain current awake healing rates
- Rebalance `Deterioration` and `Disintegration` to be negative `Fast Healer` and `Very Fast Healer` instead of nothingburgers. Rebalance `Weakening` to degrade HP at the smallest possible rate (1HP/day).

TODO:
- [X] fix mutation descriptions: `..heal 66% faster while awake...` (66% faster than zero is still zero). Removed specific information. I'll take suggestions for fluffier text.

#### Describe alternatives you've considered
Not bothering so much since there's a wounds rework in progress: there's no reason not to improve the current implementation while we're waiting for the rapture.

#### Testing
Updated test units

#### Additional context
<details>
<summary>Mutation healing rates before</summary>

|Mutation           | awake    |asleep  |
|----------------------|--------------|-----------|
|Regeneration       | 0.000200 |0.000450|
|Very Fast Healer   | 0.000066 |0.000216|
|Rapid Metabolism   | 0.000020 |0.000170|
|Fast Healer        | 0.000020 |0.000170|
|Slow Healer        | 0        |0.000075|
|Poor healer        | 0        |0.000034|
|Imperceptive Healer| 0        |0.000010|
|Weakening          | 0        |0.000075|
|Deterioration      | 0        |0.000100|
|Disintegration     |-0.000008 |0.000100|

</details>

<details>
<summary>Mutation healing rates after</summary>

|Mutation           | awake    |asleep |
|----------------------|-------------|----------|
|Regeneration       | 0.000200 |0.000250|
|Very Fast Healer   | 0.000066 |0.000150|
|Rapid Metabolism   | 0.000020 |0.000150|
|Fast Healer        | 0.000020 |0.000150|
|Slow Healer        | 0        |0.000075|
|Poor healer        | 0        |0.000033|
|Imperceptive Healer| 0        |0.000010|
|Weakening          |-0.000013 |0.000075|
|Deterioration      |-0.000020 |0.000075|
|Disintegration     |-0.000066 |0.000075|

</details>

Bandages still do double healing while sleeping. I'd be fine with removing that.

### CODE REVIEW

**Comment by andrei8l** [comment]
There are other issues that prevent these rates from working correctly, particularly the negative ones.

This is ready for review, but I'm setting it to draft so it doesn't get merged accidentally while I work on the followup.

**Comment by PatrikLundell** [comment]
Is it even possible to sleep enough to stay alive when suffering from the revised Disintegration? The recovery rate while sleeping is only just above the loss rate while awake, and you can't sleep substantially more than 8 hours per day unless suffering from a cold or taking drugs. With increased healing from bandages you can work around that, but without that boost you'd be doomed even if you never suffered any actual injuries.

**Comment by andrei8l** [comment]
> Is it even possible to sleep enough to stay alive when suffering from the revised Disintegration? The recovery rate while sleeping is only just above the loss rate while awake, and you can't sleep substantially more than 8 hours per day unless suffering from a cold or taking drugs. With increased healing from bandages you can work around that, but without that boost you'd be doomed even if you never suffered any actual injuries.

-0.000066 hp/turn from Disintegration is -5.7 hp/day so it's still pathetic and I'll increase it to be negative Regeneration instead (-17.3 hp/day). It's an end-of-the-line mutation and it shouldn't be easy to overcome.

Also keep in mind that perfect lifestyle doubles the sleep healing rate.

**Comment by andrei8l** [comment]
I've changed HP and healing to use floating point math so it can be more reliably tested on time scales that matter to the player (day-to-day fighting). I've also removed the enforced healing of 1HP/day.

After play testing, I think `Disintegration` and `Deterioration` are fine. They're pathetic on their own, but they add up with damage from fighting and etc so they're fine in the end. I've removed the `healing_awake` valid range ([-1,1]) though in case we/mods need something more hardcore.

**Comment by andrei8l** [comment]
I'm backtracking on the float change because it's far too invasive, for this PR at least. You can take a peek here https://github.com/andrei8l/Cataclysm-DDA/commit/a6237e94cec1925c006940024eda74fe3e163375.

**Comment by alp7292** [comment]
how fast healing happens in this update healing %50 health a day always felt wrong

**Comment by andrei8l** [comment]
> how fast healing happens in this update healing %50 health a day always felt wrong

This isn't a balance patch so the numbers are the same, except for the ones that were bugged. You can see the healing rates in the table above.

**Comment by bombasticSlacks** [comment]
> I'm backtracking on the float change because it's far too invasive, for this PR at least. You can take a peek here https://github.com/andrei8l/Cataclysm-DDA/commit/a6237e94cec1925c006940024eda74fe3e163375.

This looks good, worth mentioning there is currently planned work to remove hp pools for NPCs entirely, instead having damage converted to wounds and the death state be bleeding to death. There might still be value in the float conversion, but just wanted you to be aware it is something @KorGgenT is working on.
